### PR TITLE
Add meta tags supporting OneBox to document head

### PR DESCRIPTION
**Pull request by DanRathbun**
Add oEmbed / OpenGraph / Twitter Card meta elements into the HEAD element of the default HTML file.

This is in reference to: https://github.com/SketchUp/api-issue-tracker/issues/634

For more basic information, see the **reference list** at the bottom of this old API YARD Template Issue:
https://github.com/SketchUp/sketchup-yard-template/issues/18

### CODE REVIEW

**Review by Eneroth3 — CHANGES_REQUESTED** [review]

File: _layouts/default.html [hunk 1]
```diff
@@ -6,9 +6,35 @@
...
-    <title>Homepage | SketchUp Developer</title>
-    
+  <!-- TITLE -->
+    <title>SketchUp Extension UX Guidelines</title>
+
+  <!-- GENERIC META PROPERTIES -->
+    <meta name="url"   content="https://sketchup.github.io/sketchup-extension-ux-guidelines/" />
+    <meta name="image" content="https://ruby.sketchup.com/images/Ruby.svg" />
+    <meta name="title" content="SketchUp Extension UX Guidelines" />
+    <meta name="name"  content="SketchUp Ruby API Documentation" />
```
[the remark this answers is not in the document]
Comment (DanRathbun) [reply _layouts/default.html:18]: The document "title" **is**. 

The "name" property is the **site name** not the document name.

If you would like it to be language agnostic, how about "SketchUp Developer Documentation"  ?

File: _layouts/default.html [hunk 2]
```diff
@@ -6,9 +6,35 @@
...
+    <meta property="og:image:height" content="60" />
+    <meta property="og:title"     content="SketchUp Extension UX Guidelines" />
+    <meta property="og:url"       content="https://sketchup.github.io/sketchup-extension-ux-guidelines/" />
+    <meta property="og:description" content="The purpose of these guidelines is to help developers design extensions that closer align with SketchUp and to create an improved and unified experience for the end user." />
+  <!-- TWITTER CARD META PROPERTIES -->
+    <meta name="twitter:card"  content="summary"   />
+    <meta name="twitter:site"  content="@sketchup" />
+    <meta name="twitter:title" content="SketchUp Extension UX Guidelines" />
+    <meta name="twitter:description" content="The purpose of these guidelines is to help developers design extensions that closer align with SketchUp and to create an improved and unified experience for the end user." />
+    <meta name="twitter:image:src"   content="https://ruby.sketchup.com/images/Ruby.svg?s=120" />
```
[the remark this answers is not in the document]
Comment (DanRathbun) [reply _layouts/default.html]: Yes I can. (3 places.)

File: _layouts/default.html (hunk 1 above)
[the remark this answers is not in the document]
Comment (DanRathbun) [reply _layouts/default.html:18]: .. or how about "SketchUp Developer Resources" ?

File: _layouts/default.html (hunk 2 above)
[the remark this answers is not in the document]
Comment (DanRathbun) [reply _layouts/default.html]: For now ... switched to the EW logo until we get a better URL for the application logo.

File: _layouts/default.html (hunk 1 above)
[the remark this answers is not in the document]
Comment (DanRathbun) [reply _layouts/default.html]: Changed to the latter.

File: _layouts/default.html [hunk 3]
```diff
@@ -6,9 +6,35 @@
...
+  <!-- SHORTCUT ICON -->
     <link rel="shortcut icon" href="https://developer.sketchup.com/sites/developer.sketchup.com/files/Favicon.ico" type="image/vnd.microsoft.icon" />
-    <title>Homepage | SketchUp Developer</title>
-    
+  <!-- TITLE -->
+    <title>SketchUp Extension UX Guidelines</title>
+
+  <!-- GENERIC META PROPERTIES -->
+    <meta name="url"   content="https://sketchup.github.io/sketchup-extension-ux-guidelines/" />
+    <meta name="image" content="https://www.sketchup.com/sites/www.sketchup.com/files/developer/warehouse.png" />
```
Comment (thomthom) [inline _layouts/default.html:16]: These files are not stable. They can easily change, and they have in the past. Better to host them with this repo.

https://www.sketchup.com/sites/www.sketchup.com/files/developer/warehouse.png

**Review by thomthom — CHANGES_REQUESTED** [review]
It'd be better to host the image assets in this repo to make it resilient from changes made to the other sites.

File: _layouts/default.html (hunk 3 above)
Comment (Eneroth3) [reply _layouts/default.html:16]: Good point!

Do you have a link to the general SU logo? As of now extensibility doesn't have any icon, and this project isn't a part of the Extension Warehouse.

Comment (DanRathbun) [reply _layouts/default.html:16]: > _These files are not stable. They can easily change, and they have in the past. Better to host them with this repo._

I do not understand this statement. Please be specific. **What files? What URLs change ?**

The idea is that when a URL is pasted in a Discourse forum (anywhere), in Twitter, in FaceBook, in wherever oEmbed or OpenGraph one boxes are supported, that the page referenced has the meta tags that have the properties to populate the onebox or twitter card or whatever.

So a stable URL is needed for pasting into other sites. I do not care **what** it is. As long as it's stable. (Ie, there are numerous links previously posted to the forum that have no redirects and are now "dead links".

> _Do you have a link to the general SU logo?_

I could not find a square application logo in png format. 
All the images I found have a gray "SketchUp" to the right of the logo.

Is there a logo only image embedded within the executable ? 
(Perhaps used for message boxes? or the installer?)

> _As of now extensibility doesn't have any icon, and this project isn't a part of the Extension Warehouse._

I was under the impression that the new blue X logo stood for extensibility (ie extensions) and that EW was generally a part of SketchUp's extensibility.

Comment (DanRathbun) [reply _layouts/default.html:16]: Do you wish to also put the favicon in this repo's "assets" folder ?

File: _layouts/default.html (hunk 2 above)
[the remark this answers is not in the document]
Comment (DanRathbun) [reply _layouts/default.html]: I can change the logo URL as soon as you can copy it somewhere beneath the "assets" folder.
Same for the favicon if Thomas was also referring to that location also being "fragile".

File: _layouts/default.html (hunk 3 above)
Comment (DanRathbun) [reply _layouts/default.html]: Logo has been changed to Mark's logo: copied into assets folder.

**Comment by DanRathbun** [comment]
Five months and counting ...

**Comment by thomthom** [comment]
`master` has been renamed to `main`. Rename your fork's `master` to `main` and update your local branch to track `main`.

GitHub should have been displaying a notice to forks about that.

**Comment by thomthom** [comment]
If you don't have any other branches you need then that would work as well.

I use UI tools to work with Git. GitKraken or GitForks. Makes renaming and resetting branches so much easier.

**Pull request merged**